slimt/Annotation.cc

```diff
@@ -43,6 +43,9 @@ void AnnotatedText::append_ending_whitespace(std::string_view whitespace) {
   text.append(whitespace.data(), whitespace.size());
   annotation.token_begin_.back() = text.size();
 }
+void AnnotatedText::update_annotation(const std::vector<Range> &words) {
+  annotation.update_words(words);
```

Comment:
As a general rule of thumb, `cls.do_x(x)` -> `cls.do(x)` leads to more pleasant code. In this case see the difference (at call-site):

```diff
-annotation.update_words(words)
+annotation.update(words)

-annotated.update_annotation(annotation)
+annotated.update(annotation)
```

I realize it's not followed, but let's try to keep it that way. Classes are _nouns_. Functions are _verbs_. Keep an optimization objective of brevity in names.